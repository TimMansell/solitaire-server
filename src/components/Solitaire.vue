<template>
  <div class="solitaire">
    <Board />
    <GameOverlay v-if="!hasMoves">
      <div v-if="isGameWon" data-test="game-won">Congratulations, you win!</div>
      <div v-if="isGameLost" data-test="game-lost">Sorry, no more Moves!</div>
    </GameOverlay>
    <Rules />
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import Board from '@/components/Board.vue';
import GameOverlay from '@/components/GameOverlay.vue';
import Rules from '@/components/Rules.vue';
// import aces from '../../tests/fixtures/boards/fullFoundation.json';

export default {
  name: 'Home',
  components: {
    Board,
    GameOverlay,
    Rules,
  },
  computed: {
    ...mapGetters(['hasMoves', 'isGameWon', 'isGameLost']),
  },
  async created() {
    await this.initUser();
    await this.initGame();

    // Force cypress to wait until async functions have loaded.
    if (window.Cypress) {
      window.appReady = true;
    }

    // if (process.env.NODE_ENV === 'development') {
    //   this.setBoardAndFoundation(aces);
    // }
  },
  methods: {
    ...mapActions(['initUser', 'initGame', 'setBoardAndFoundation']),
  },
};
</script>

<style scoped lang="scss">
.solitaire {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0.5rem 0;
}
</style>
