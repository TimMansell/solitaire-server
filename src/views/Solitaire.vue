<template>
  <div class="solitaire">
    <Board />
    <div v-if="isGameWon">
      <Winner />
    </div>
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import Board from '@/components/Board.vue';
import Winner from '@/components/Winner.vue';
// import emptyColumn from '../../tests/fixtures/boards/emptyColumn.json';
// import aces from '../../tests/fixtures/decks/moveAcetoAces.json';

export default {
  name: 'Home',
  components: {
    Board,
    Winner,
  },
  data() {
    return {
      gameWon: true,
    };
  },
  computed: {
    ...mapGetters(['isGameWon']),
  },
  mounted() {
    this.initGame();
    // this.$store.dispatch('setTestBoard', emptyColumn);
    // this.$store.dispatch('dealTestCards', aces);
  },
  methods: {
    ...mapActions(['initGame']),
  },
};
</script>

<style scoped lang="scss">
.solitaire {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0.5rem 0;
}
</style>
