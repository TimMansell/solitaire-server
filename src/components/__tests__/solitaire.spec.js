import { shallowMount } from '@vue/test-utils';
import Solitaire from '@/components/Solitaire.vue';

const mocks = {
  $store: { dispatch: jest.fn() },
};

describe('Solitaire.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Solitaire, {
      mocks,
      computed: {
        showStats: () => true,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('shows stats overlay', () => {
    const wrapper = shallowMount(Solitaire, {
      mocks,
      computed: {
        showStats: () => true,
      },
    });

    expect(wrapper.find('[data-test="stats-overlay"]').exists()).toBe(true);
  });

  it('does not shows stats overlay', () => {
    const wrapper = shallowMount(Solitaire, {
      mocks,
      computed: {
        showStats: () => false,
      },
    });

    expect(wrapper.find('[data-test="stats-overlay"]').exists()).toBe(false);
  });
});
