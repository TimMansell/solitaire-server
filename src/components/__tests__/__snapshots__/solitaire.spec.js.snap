// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Solitaire.vue matches snapshot 1`] = `
<div class="solitaire">
  <board-stub></board-stub>
  <gamestate-stub></gamestate-stub>
  <rules-stub></rules-stub>
  <statsoverlay-stub data-test="stats-overlay"></statsoverlay-stub>
</div>
`;
